regolith/helpers/l_contactshelper.py

```diff
@@ -0,0 +1,183 @@
+    subpi.add_argument("-y", "--year", help='match people to years')
+    subpi.add_argument("-r", "--misc", help='match people to miscellaneous notes')
+    '''
+    subpi.add_argument("-v", "--verbose", action="store_true", help='increase verbosity of output')
+    subpi.add_argument("-l", "--lead",
+                       help="Filter milestones for this project lead"
+                       )
+    subpi.add_argument("-p", "--person",
+                       help="Filter milestones for this person whether lead or not"
+                       )
+    subpi.add_argument("-s", "--stati", nargs="+",
+                       help=f"List of stati for the project that you want returned,"
+                            f"from {ALLOWED_STATI}.  Default is proposed and started"
+                       )'''
+    return subpi
+
+
+class ContactsListerHelper(SoutHelperBase):
+    """Helper for listing upcoming (and past) projectum milestones.
+
+       Projecta are small bite-sized project quanta that typically will result in
+       one manuscript.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = HELPER_TARGET
+    needed_dbs = [f'{TARGET_COLL}']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        if "groups" in self.needed_dbs:
+            rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        try:
+            if not rc.database:
+                rc.database = rc.databases[0]["name"]
+        except:
+            pass
+        colls = [
+            sorted(
+                all_docs_from_collection(rc.client, collname), key=_id_key
+            )
+            for collname in self.needed_dbs
+        ]
+        for db, coll in zip(self.needed_dbs, colls):
+            gtx[db] = coll
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+
+    def sout(self):
+        rc = self.rc
+        contacts = []
+        '''
+        if rc.lead and rc.person:
+            raise RuntimeError(f"please specify either lead or person, not both")'''
+        for contact in self.gtx["contacts"]:
+            
+            #Handling names
+            if rc.name:
+                list_add = False
+                for nam in rc.name.split(","):
+                    if (contact.get('name').lower()).find(nam.lower()) != -1:
+                        contacts.append(contact)
+                        list_add = True
+                if list_add:
+                    continue
+                
+            #Handling institutions
+            if rc.inst and (contact.get('institution').lower()).find(rc.inst.lower()) != -1:
+                contacts.append(contact)
+                continue
+            
+            #Handling days/months/years
+            if rc.month and rc.year and rc.day:
```

Comment:
see my comments above.  This needs a major refactor.  This seems like a good place to use is-current.....  Use dateutils for the date deltas.   Look how it is done in ``get_dates()`` and ``is_current()`` in ``dates.py`` and reuse those well tested functions as much as you can.